### PR TITLE
[4.0] [a11y] aria-required

### PR DESCRIPTION
**Pull request by brianteeman**
The attribute aria-required is redundant in html5 if the required attribute is already set

http://html5doctor.com/on-html-belts-and-aria-braces/

There is one js file changes @dgrammatiko please check

There is one file still to update that I would like some help on components\com_contact\layouts\joomla\form\renderfield.php

### CODE REVIEW

**Review by dgrammatiko — APPROVED** [review]

**Comment by wilsonge** [comment]
What help do you need in that layout?

**Comment by brianteeman** [comment]
line 43. i dont understand that code enough to make the required change

**Comment by wilsonge** [comment]
So that is trying to do a weird check if the input field is required by checking for the `aria-required="true"` in the input string or if there is a class `required` in the input string. Given the thing can't have an aria-attribute anymore you're safe to remove that check - but I don't know if we have required class either anymore or if we've just left it to the attribute - if it's now the required attribute then we're going to need to replace that check (can be validated in the frontend by seeing if that optional string is showing)

**Comment by brianteeman** [comment]
@wilsonge I can not even see where that is used

**Comment by zwiastunsw** [comment]
@brianteeman  is right if the required attribute is used, then the aria-required attribute is redundant (This article refers to the role attributes, but today the screen readers, NVDA and JAWS certainly announce the required attribute, so no additional aria is needed).

**Comment by Quy** [comment]
Here are vardumps:

$match[1]: 
`string(21) "form-control required"`

$input:
```
<input
		type="text"
		name="jform[contact_name]"
		id="jform_contact_name"
		value=""
				class="form-control required" size="30"        required      >
```

so this change should work:
`$required      = !empty($match[1]) && strpos($match[1], 'required') !== false;`

**Comment by brianteeman** [comment]
@quy @wilsonge I dont like removing any code unless i can see what its doing (or not)
I would assume that this code was adding the aria attribute but its not. In which case what is the code doing - if anything at all

**Comment by Quy** [comment]
The code appends `(optional)` to the label when not required.

Create a mail custom field with/without required attribute.
On the frontend, load a contact form page.

**Comment by brianteeman** [comment]
@quy thanks I hadnt looked at custom fields. However it still never added the aria required ??

**Comment by Quy** [comment]
Without PR, `aria-required="true"` is present.
```
<input
		type="text"
		name="jform[contact_name]"
		id="jform_contact_name"
		value=""
				class="form-control required" size="30"        required aria-required="true"      >
```

With PR, no ``aria-required="true"``, therefore, it should be safe to remove `strpos($input, 'aria-required="true"') !== false` check.
```
<input
		type="text"
		name="jform[contact_name]"
		id="jform_contact_name"
		value=""
				class="form-control required" size="30"        required      
```

File: administrator/components/com_redirect/tmpl/links/default_batch_body.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
 		<div class="control-group col-md-12">
             <p><?php echo JText::sprintf('COM_REDIRECT_BATCH_TIP', $separator); ?></p>
 			<div class="controls">
-				<textarea class="col-md-12" rows="10" aria-required="true" value="" id="batch_urls" name="batch_urls"></textarea>
+				<textarea class="col-md-12" rows="10" value="" id="batch_urls" name="batch_urls"></textarea>
```
Comment (Quy) [inline administrator/components/com_redirect/tmpl/links/default_batch_body.php]: Add `required`

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 5510c5ef9f460abd93bbd3a1ff2baa10f7bdd4b7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20974">issues.joomla.org/tracker/joomla-cms/20974</a>.</sub>

**Comment by brianteeman** [comment]
Can we get one more test please so this can be merged

**Comment by zwiastunsw** [comment]
I have tested this item :white_check_mark: successfully on 592b80f93562e2a74893a753bd507709e920caf6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20974">issues.joomla.org/tracker/joomla-cms/20974</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after successful tests.

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**